arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -921,6 +921,35 @@ uint64_t MaintenanceFeature::shardVersion(std::string const& shname) const {
 
 uint64_t MaintenanceFeature::incShardVersion(std::string const& shname) {
   MUTEX_LOCKER(guard, _versionLock);
+
+  uint64_t v = 0;
+  using namespace std::chrono;
+  using clock = steady_clock;
+  auto timeout = duration<double>(600.0);
+  auto stoppage = clock::now();
+  auto snooze = milliseconds(100);
+
+  // Acquire current version from agency and wait for it to have been dealt
+  // with in local current cache. Any future current version will do, as
+  // the version is incremented by the leader ahead of getting here on the
+  // follower.
+  while (!server().isStopping() && clock::now() < stoppage) {
```

Comment:
`stoppage` is set to `clock::now()` a few lines above and never updated. Shouldn't `stoppage` rather be set to something like `clock::now() + timeout?